sklearn/linear_model/_coordinate_descent.py

```diff
@@ -903,6 +904,13 @@ def fit(self, X, y, sample_weight=None, check_input=True):
             self.normalize, default=False, estimator_name=self.__class__.__name__
         )
 
+        self.alpha = check_scalar(
```

Comment:
I know that `check_scalar ` does not actually change the input, but setting `self.alpha` here, gives the appearance that `self.alpha` is being changed. I suggest we remove the `self.alpha = ` part.

```suggestion
        check_scalar(
```

This comment applies to the rest of the `check_scalar` calls.

Reply:
Done, thanks!